nilearn/plotting/displays.py

```diff
@@ -227,7 +228,17 @@ def transform_to_2d(self, data, affine):
 
         """
         max_axis = 'xyz'.index(self.direction)
-        maximum_intensity_data = np.abs(data).max(axis=max_axis)
+        
+        if self._plot_negative:
+            new_shape = list(data.shape)
+            del new_shape[max_axis]
+            a1, a2 = np.indices(new_shape)
+            inds = [a1, a2]
+            inds.insert(max_axis, np.abs(data).argmax(axis=max_axis))
```

Comment:
I may be a little dense but the lines above are not very clear to me. Could you add a comment or two?